### DISCUSSION TITLE
How to interact with file system and run processes using Quasar?

### DISCUSSION THREAD

**Discussion by FlogramMatt** [quasarframework/quasar#9827] [CLI - Electron mode]
I'm looking to build an IDE that uses Quasar, I need to be able to save, load, and query the file system to display files for users.

Additionally, I either need to call C++ libraries from Quasar, which I know is possible but looks complicated, OR run a process from Quasar in order to allow users to run my code.

Are these both possible with Quasar? Can someone please point me in the right direction? Don't see much googling but seems like it must be possible.. maybe electron specific?

**Comment by smolinari** [comment]
@FractalMatt - Your concern is most definitely at the Electron/ NodeJS level. Quasar is just the UI side of your application. "The backend" is completely up to you. 😃 

Scott

**Comment by hawkeye64** [comment]
@FractalMatt You can look at this repo (uses Quasar v1.x): https://github.com/hawkeye64/electron-quasar-file-explorer and you can extrapolate some concepts of the file system access.
As for the C++, either bind it to Node (difficult) or convert it to WASM (I have no experience in either)
